File: persistence-test/src/test/java/org/mariella/test/model/File.java
package org.mariella.test.model;

import jakarta.persistence.*;

@jakarta.persistence.Entity
@Table(name = "RESOURCE")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorValue("File")
public class File extends Resource {
    private long size;

    @Column(name = "SIZE")
    public long getSize() {
        return size;
    }

    public void setSize(long size) {
        propertyChangeSupport.firePropertyChange("size", this.size, this.size = size);
    }

}

File: persistence-kotlin/src/test/kotlin/org/mariella/test/entities/File.kt
package org.mariella.test.entities

import javax.persistence.*
import javax.persistence.Entity

@Entity
@Table(name = "file_node")
@PrimaryKeyJoinColumn(name = "id", referencedColumnName = "id")
@Inheritance(strategy = InheritanceType.JOINED)
@DiscriminatorValue("FI")
class File : Resource() {
    override fun isContainer(): Boolean {
        return false
    }
}